cranelift/codegen/src/isa/zkasm/inst/emit.rs

```diff
@@ -1366,10 +1368,24 @@ impl MachInstEmit for Inst {
                 put_string(&format!("$ => {} :{opcode}\n", reg_name(rd.to_reg())), sink);
 
                 if cc == IntCC::NotEqual {
-                    put_string("1 => B\n", sink);
+                    let one: u64 = if ty == types::I32 {
+                        1
+                    } else if ty == types::I64 {
+                        4294967296
+                    } else {
+                        todo!()
+                    };
+                    put_string(&format!("{one} => B\n"), sink);
                     put_string(&format!("$ => {} :XOR\n", reg_name(rd.to_reg())), sink);
                 }
 
+                // Result of comparing operations in wasm for both i32 and i64
+                // is i32. So, we need to multiply it by 2**32
+                put_string("4294967296n => B\n", sink);
+                put_string("0 => D\n", sink);
+                put_string("0 => C\n", sink);
+                put_string("${A * B} => A :ARITH\n", sink);
```

Comment:
My intuition suggests using jumps here would be faster if only by the virtue of forcing fewer register spills.

Reply:
(this is non-blocking for landing)